unity/unity-embed-host/UnsafeExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace Unity.CoreCLRHelpers;
+
+/// <summary>
+/// These extensions are specifically for transitioning to/from our current native representation
+/// These are not GC safe and will change in the future
+/// </summary>
+static class UnsafeExtensions
+{
+    public static object ToManagedRepresentation(this nint intPtr)
+        => Unsafe.As<nint, object>(ref intPtr);
+
+    public static nint ToNativeRepresentation(this object obj)
+        => obj.ToNativeRepresentation<nint>();
+
+    public static T ToNativeRepresentation<T>(this object obj)
```

Comment:
Nit, does any code need this public generic version?

Reply:
No.  Now that you mention it, I don't need it's needed at all.  I removed it.

Reply:
Also nit: "Representation" is a little wordy. Could we have `ToManaged` and `ToNative` ?

Reply:
I'll add my opinion too ;-) maybe there is a better word, but I am okay with more verbosity for these specific APIs as I want to make sure whoever is calling them understands what the implications are.

Reply:
I went wordy for the reason @joncham brought up.   I'm happy to change it to something else, but I do a agree that `ToManaged` and `ToNative` are too simple